tests/test_superruns.py

```diff
@@ -0,0 +1,247 @@
+        assert len(chunks) == 1
+        chunk = chunks[0]
+        assert chunk['run_id'] == self.superrun_name
+        assert chunk['first_time'] == subrun_data['time'].min()
+        assert chunk['last_endtime'] == np.max(strax.endtime(subrun_data))
+
+    def test_select_runs(self):
+        self.context.select_runs()
+        self.context.make(self.superrun_name,
+                          'records',
+                          _check_lineage_per_run_id=False
+                          )
+        df = self.context.select_runs(available=('records',))
+        assert self.superrun_name in df['name'].values
+
+    def test_superrun_chunk_and_meta(self):
+        """
+        Superrun chunks and meta data should contain information about
+        its constituent subruns.
+        """
+        self.context.make(self.superrun_name,
+                          'records',
+                          _check_lineage_per_run_id=False
+                          )
+
+        meta = self.context.get_meta(self.superrun_name, 'records')
+
+        n_chunks = 0
+        superrun_chunk = None
+        for chunk in self.context.get_iter(self.superrun_name, 'records'):
+            superrun_chunk = chunk
+            n_chunks += 1
+
+        assert len(meta['chunks']) == n_chunks == 1
+        assert meta['chunks'][0]['subruns'] == superrun_chunk.subruns
+
+        for subrun_id, start_and_end in superrun_chunk.subruns.items():
+            rr = self.context.get_array(subrun_id, 'records')
+            # Tests below only true for records as we have not rechunked yet.
+            # After rechunking in general data start can be different from chunk start
+            mes = f'Start time did not match for subrun: {subrun_id}'
+            assert rr['time'].min() == start_and_end['start'], mes
+            mes = f'End time did not match for subrun: {subrun_id}'
+            assert np.max(strax.endtime(rr)) == start_and_end['end'], mes
+
+    def test_merger_plugin_subruns(self):
+        """
+        Tests if merge plugins for subruns work. This test is needed to
+        ensure that superruns do not interfer with merge plguns.
+        (both are using underscores as identification).
+        """
+        rr = self.context.get_array(self.subrun_ids, ('records', 'records_extension'))
+        p = self.context.get_single_plugin(self.subrun_ids[0], 'records_extension')
+        assert np.all(rr['additional_field'] == p.config['some_additional_value'])
+        
+    def test_rechnunking_and_loading(self):
+        """
+        Tests rechunking and loading of superruns with multiple chunks.
+        
+        The test is required since it was possible to run into race conditions with 
+        chunk.continuity_check in context.get_iter and transform_chunk_to_superrun_chunk 
+        in storage.common.Saver.save_from.
+        """
+        self.context.set_config({'recs_per_chunk': 100000}) # Make chunks > 200 MB
```

Comment:
This can make the rest unnecessarily slow or heavy on I/O, perhaps set the max size in the `st._plugin_class_registry['records'].chunk_target_size_mb =1`